File: rolan_code/win/Libraries/JavaScriptInterface/NativeCall+notification.cpp
//
//  NativeCall+notification.cpp
//
//  Copyright 2018 Roland Corporation. All rights reserved.
//

#include "NativeCall.h"

static LPCWSTR OBJ_NAME = L"notification";

enum {
	dispid_notification_token,
	dispid_notification_badge,
};

NativeCall_notification::NativeCall_notification(NativeObjects* objs) : NativeCall(objs)
{
	set(L"token", dispid_notification_token);
	set(L"badge", dispid_notification_badge);
}

LPCWSTR NativeCall_notification::InterfaceName() const { return OBJ_NAME; }

void NativeCall_notification::Dispatch(UINT dispid, const CefV8ValueList& args, CefRetValue& ret)
{
	switch (dispid) {
		case dispid_notification_token:
		{
			ret = L""; /* no device token */
			break;
		}
		case dispid_notification_badge:
		{
			ret = 0; /* always return 0 */
			break;
		}
	}
}

File: rolan_code/mac/Libraries/JavaScriptInterface/NativeCall+notification.m
//
//  NativeCall+notification.m
//
//  Copyright 2018 Roland Corporation. All rights reserved.
//

#import <TargetConditionals.h>
#if TARGET_OS_IPHONE
#import <UIKit/UIKit.h>
#else
#import <Cocoa/Cocoa.h>
#endif
#import "NativeCall.h"

//static NSString *const OBJ_NAME = @"notification";

NSString *const DEVICE_TOKEN_KEY = @"device-token";

@interface NativeCall (notification) @end

@implementation NativeCall (notification)

- (NSString *)$$notification_token:(NSString *)param1 :(NSString *)param2
{
    NSString *token = [[NSUserDefaults standardUserDefaults] stringForKey:DEVICE_TOKEN_KEY];
#ifdef DEBUG
    NSLog(@"apns device token: %@", token);
#endif
    return_s(token);
}

- (NSString *)$$notification_badge:(NSString *)param1 :(NSString *)param2
{
#if TARGET_OS_IPHONE
	if (param1) {
		[UIApplication sharedApplication].applicationIconBadgeNumber = [param1 intValue];
	}
	return_d([UIApplication sharedApplication].applicationIconBadgeNumber);
#else
	return_d(0); /* always return 0 */
#endif
}

@end
